From 2d2953481cedc020934e461cf5d465afd1e1e547 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Tue, 10 Mar 2020 12:40:22 -0700
Subject: [PATCH 24/27] Revert "pid: make ENOMEM return value more obvious"

Take the updated version of this patch to avoid a clang warning.

This reverts commit 8deb24dcb89cb390110d2ccac830a84d1ab5cee4.

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 kernel/pid.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/kernel/pid.c b/kernel/pid.c
index d4e0622e8b85..cf22d84ab1a8 100644
--- a/kernel/pid.c
+++ b/kernel/pid.c
@@ -244,14 +244,6 @@ struct pid *alloc_pid(struct pid_namespace *ns, pid_t *set_tid,
 		tmp = tmp->parent;
 	}
 
-	/*
-	 * ENOMEM is not the most obvious choice especially for the case
-	 * where the child subreaper has already exited and the pid
-	 * namespace denies the creation of any new processes. But ENOMEM
-	 * is what we have exposed to userspace for a long time and it is
-	 * documented behavior for pid namespaces. So we can't easily
-	 * change it even if there were an error code better suited.
-	 */
 	retval = -ENOMEM;
 
 	get_pid_ns(ns);
-- 
2.26.0.rc1

